### PR TITLE
Have removed sessions in Category, Drilldown and Question Controllers

### PR DESCRIPTION
**Pull request by ProsperChiduku**

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ def form
     user = User.new(email: params["email"])
     user.save
     session[:user]=user.id
```
Comment (aaronmyatt) [inline app/controllers/home_controller.rb]: So, for your information, it is definitely good practice to keep a reference to the user in the session, but we can abstract this sort of common pattern into helpers, here's how MV Home is doing it: https://github.com/mindvalley/MVA_Home/blob/fdffdd751b43519d2cdafd8ecf7d7bf61903dab4/app/helpers/auth0_helper.rb

We can do the same once we implement authentication.

File: app/controllers/questions_controller.rb [hunk 2]
```diff
@@ -13,11 +13,11 @@ def index
   end
 
   def form
-    question_session = session[:questions] = params[:questions]
+    question_session = params[:questions]
```
Comment (aaronmyatt) [inline app/controllers/questions_controller.rb]: Can we rename this variable?, maybe `question_params`?

File: app/controllers/recommendation_controller.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 class RecommendationController < ApplicationController
   def index
-    @category = session[:category]
-    @drilldown = session[:drilldown]
-    @questions = session[:questions]
+    @category = Category.last(4).pluck(:name)
```
Comment (aaronmyatt) [inline app/controllers/recommendation_controller.rb]: So, this isn't gonna work for us in the long term. Eventually there will be two people who perform the quiz at the same time, someone will see the wrong data.

Let's update the backlog with a ticket to connect the cateogry, drilldown and question choices to the users email

File: app/views/category/index.html.erb [hunk 4]
```diff
@@ -33,7 +29,7 @@
     <li>
     <div id="check-button">
       <label>
-         <%= check_box_tag x.name,x.id  %> 
+         <%= check_box_tag "category[]", x.name  %>
```
Comment (aaronmyatt) [inline app/views/category/index.html.erb]: rename the `x` please

**Review by aaronmyatt — CHANGES_REQUESTED** [review]

**Comment by aaronmyatt** [comment]
All in all, really great, let's tidy some of the parameter naming, and also gave a little food for thought,.

File: app/controllers/questions_controller.rb (hunk 2 above)
Comment (pavikunusilin) [reply app/controllers/questions_controller.rb]: DONE.

File: app/views/category/index.html.erb (hunk 4 above)
Comment (pavikunusilin) [reply app/views/category/index.html.erb]: DONE.

**Pull request merged**